app/lib/efile/nj/nj1040.rb

```diff
@@ -67,21 +78,10 @@ def calculate_line_40a
         is_mfs ? (property_tax_paid / 2.0).round : property_tax_paid.round
       end
 
-      def total_exemption_amount
-        0
+      def calculate_line_13
+        calculate_line_6 + calculate_line_7 + calculate_line_8
```

Comment:
[pebble] since some of the calculations rely on each other and need to be done in a procedural fashion we follow the practice of only calling the calculate methods when we are setting the lines in the calculate method at the top and then calling the set line everywhere else so that it would look like this: 
```suggestion
        line_or_zero(:NJ1040_LINE_6) +  line_or_zero(:NJ1040_LINE_7) +  line_or_zero(:NJ1040_LINE_8) 
```

Reply:
Thanks for letting me know about this! Can you elaborate on the danger of leaving this as a method call? Happy to change, just trying to understand this hazard better.

Reply:
Yeah I wish I could find the example of where we ran into this problem years ago (maybe I'll look for it if it doesn't chew up too much time and send it over) but basically we've run into problems where calling a calculation multiple times would change its value which is why we only call it once and run through the lines in order. I think there could also potentially be infinite loop/recursion issues hypothetically since lines depend on each other.